src/com/ansorgit/plugins/bash/editor/EnterInStringLiteralHandler.java

```diff
@@ -48,6 +48,10 @@ public Result preprocessEnter(@NotNull PsiFile file, @NotNull Editor editor, @No
             return Result.Continue;
         }
 
+        // as advised in the JavaDoc of
+        // com.intellij.codeInsight.editorActions.enter.EnterHandlerDelegate.preprocessEnter
+        PsiDocumentManager.getInstance(file.getProject()).commitDocument(editor.getDocument());
```

Comment:
When you moved that commit out from the if statement it performs even worse :)
It's better to keep some checks with document and commit it after if needed. 

Just open a ~1mb file and try to type inside it trying to invoke that handler – that shouldn't be pleasant.

Reply:
@ignatov If I'm not mistaken the IntelliJ SDK suggests to do it this way, see https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/codeInsight/editorActions/enter/EnterHandlerDelegate.java#L39

To me this means that I have to do this because I'm accessing the PSI.
Let me know if I'm misinterpreting this.

Reply:
> If any operations with PSI are needed including a search for PSI elements, the document must be committed first to update the PSI.

Returning from the handler earlier without commit will tremendously speed up typing.

Reply:
@ignatov Okay, returning early now when uncomitted. I think that this could result in line wrappings not inserted when editing (and when uncomitted), but that's probably better than comitting while typing.